src/frontend/src/main.js

```diff
@@ -1,21 +1,115 @@
-import * as THREE from 'three';
+import * as THREE from "three";
 
 const scene = new THREE.Scene();
-const camera = new THREE.PerspectiveCamera(75, window.innerWidth / window.innerHeight, 0.1, 1000);
+const camera = new THREE.PerspectiveCamera(
+  75,
+  window.innerWidth / window.innerHeight,
+  0.1,
+  1000
+);
 const renderer = new THREE.WebGLRenderer();
 renderer.setSize(window.innerWidth, window.innerHeight);
 
 document.body.appendChild(renderer.domElement);
 
 const geometry = new THREE.BoxGeometry(1, 1, 1);
 const material = new THREE.MeshBasicMaterial({ color: 0x00ff00 });
-const cube = new THREE.Mesh(geometry, material); scene.add(cube);
+const cube = new THREE.Mesh(geometry, material);
+scene.add(cube);
 camera.position.z = 5;
 
+function onWindowResize() {
+  camera.aspect = window.innerWidth / window.innerHeight;
+  camera.updateProjectionMatrix();
+  renderer.setSize(window.innerWidth, window.innerHeight);
+}
+
 function animate() {
-    cube.rotation.x += 0.01;
-    cube.rotation.y += 0.01;
-    renderer.render(scene, camera);
+  onWindowResize();
+
+  cube.rotation.x += 0.01;
+  cube.rotation.y += 0.01;
+  renderer.render(scene, camera);
 }
 
 renderer.setAnimationLoop(animate);
+
+const cardCounts = {};
+
+function initializeCardCounts(charId, tokenTypes) {
+  if (!cardCounts[charId]) {
+    cardCounts[charId] = {};
+  }
+  tokenTypes.forEach((_, index) => {
+    cardCounts[charId][index + 1] = 0;
+  });
+}
+
+function updateTokenCount(token) {
+  //This is the token clicking hook here you can change and get info of the selected token ,inc or dec the count corresponding count to it.
+  //Remember to update the cardCounts data structure accordingly for smoother data storing.
+  const character = token.closest("#character-block");
+  const characterId = character.getAttribute("char-id");
+  const p = token.nextElementSibling;
+  const tokenId = token.getAttribute("token-no");
+  cardCounts[characterId][tokenId] += 1;
+  console.log(`The new value is ${cardCounts[characterId][tokenId]}`)
+  console.log("Decreament here and add changes according to yourself.");
+  p.textContent = `x${cardCounts[characterId][tokenId]}`;
+}
+
+function tooltip(tokenType) {
+  if (tokenType === "square")
+    return `This is a trap.`
+  else
+    return `This is nothing.`
+}
+
+function createCard(parent, charId, imgSrc, imgAlt, tokenTypes = []) {
```

Comment:
Does this function need to define each element of the whole character and card structure one by one every time we create any of them or can we make it so each main component inside is defined once and reused.